### ISSUE TITLE
panic against nghttp2.org

### ISSUE THREAD

**Issue by agrover** [mozilla/neqo#264]
`RUST_BACKTRACE=1 RUST_LOG=debug,neqo_transport::dump ./target/debug/neqo-client https://nghttp2.org:4433/`

**Comment by agrover** [comment]
http frame parsing issue. @ddragana maybe your refactor could also squash this one?

```
0s364ms DEBUG [TransactionClient 0] A new frame has been received.
thread 'main' panicked at 'Read should succeed: HttpFrameUnexpected',
```

Should be very reproducible.

**Comment by ddragana** [comment]
I know about this one. It will be fixed soon.
The server send trailers and for time been, until I implement trailers, I interpreted it as a HttpFrameUnexpected. It is easy to fix. I may as well add trailer now. it is not a lot of work.

**Comment by ddragana** [comment]
Fixed by #444
